rasa/core/brokers/kafka.py

```diff
@@ -194,10 +196,24 @@ def _publish(self, event: Dict[Text, Any]) -> None:
         else:
             partition_key = None
 
+        headers = [
+            (
+                "RASA_ENVIRONMENT",
+                bytes(self.rasa_environment, encoding=DEFAULT_ENCODING),
+            )
+        ]
+
         logger.debug(
-            f"Calling kafka send({self.topic}, value={event}, key={partition_key!s})"
+            f"Calling kafka send({self.topic}, value={event},"
+            f" key={partition_key!s}, headers={headers})"
         )
-        self.producer.send(self.topic, value=event, key=partition_key)
+
+        self.producer.send(self.topic, value=event, key=partition_key, headers=headers)
 
     def _close(self) -> None:
         self.producer.close()
+
+    @rasa.shared.utils.common.lazy_property
+    def rasa_environment(self) -> Optional[Text]:
+        """Get value of the `RASA_ENVIRONMENT` environment variable."""
```

Comment:
@m-vdb in case you're after extremely nit-picky comments 😂 
```suggestion
        """Gets the value of the `RASA_ENVIRONMENT` environment variable."""
```

Reply:
we don't really adjust the code in these PRs, it should have been done in the original PR 😬  (this is documented on `main` [here](https://github.com/RasaHQ/rasa/blob/main/.github/PULL_REQUEST_AUTOMATIC_TEMPLATE.md))

Reply:
but thanks for noticing!